catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableFileSystemFileConsumer.java

```diff
@@ -40,7 +39,7 @@ protected boolean doPoll(String sha1) {
     if (observer != null) {
       observer.addListener(listener);
       observer.checkAndNotify();
-      observer.removeListener(listener);
+      observer.removeListener();
```

Comment:
I believe the original reasoning for this is because this class implements Serializable. So you could remove any listener who doesn't implement that before writing it to disk.